src/configuration/configuration_manager.cpp

```diff
@@ -208,9 +224,14 @@ void configuration_manager::remove_config(const configuration_change_spec &cfg)
     for (const auto &id : cfg.user_rules) { global_config_.user_rules.erase(id); }
     for (const auto &id : cfg.rule_filters) { global_config_.rule_filters.erase(id); }
     for (const auto &id : cfg.input_filters) { global_config_.input_filters.erase(id); }
-    for (const auto &id : cfg.overrides_by_id) { global_config_.overrides_by_id.erase(id); }
-    for (const auto &id : cfg.overrides_by_tags) { global_config_.overrides_by_tags.erase(id); }
+    for (const auto &id : cfg.rule_overrides_by_id) {
+        global_config_.rule_overrides_by_id.erase(id);
+    }
+    for (const auto &id : cfg.rule_overrides_by_tags) {
+        global_config_.rule_overrides_by_tags.erase(id);
+    }
     for (const auto &id : cfg.processors) { global_config_.processors.erase(id); }
+    for (const auto &id : cfg.processor_overrides) { global_config_.processor_overrides.erase(id); }
```

Comment:
This is not an issue for the processor overrides, but for types with id that were **not** added due to e.g. duplicated id, do you ensure that their original counterpart (the items with the first-encountered duplicate id) are not removed when the config with the duplicate ids is removed?

Reply:
Added in https://github.com/DataDog/libddwaf/pull/400/commits/0843d37dac0422dcaf4a1997a5a379ae94262bbd